### PR TITLE
Add initialize function for EPS sensors

### LINKED ISSUE #212: Implement initialize function for EPS sensors

### PR DESCRIPTION
**Pull request by danielschloms**
### Description
This PR adds functionality to initialize the EPS ADCs for the COBC, including the necessary pin definitions.

Fixes #212

### CODE REVIEW

**Comment by danielschloms** [comment]
There are still 2 unresolved issues:
- This cannot be tested until the COBC is available again.
- The FRAM shares the same SPI, but uses a higher baudrate. This does not seem to be okay, as RODOS does not change the baudrate before transferring data, and the max. baudrate in the datasheet is 10 MHz (FRAM uses 12 MHz).

File: Sts1CobcSw/Periphery/Eps.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#include <Sts1CobcSw/Hal/GpioPin.hpp>
+#include <Sts1CobcSw/Hal/IoNames.hpp>
+#include <Sts1CobcSw/Hal/Spi.hpp>
+#include <Sts1CobcSw/Periphery/Eps.hpp>
+#include <Sts1CobcSw/Serial/Byte.hpp>
+#include <Sts1CobcSw/Utility/Span.hpp>
+
+#include <rodos_no_using_namespace.h>
+
+namespace sts1cobcsw::periphery::eps
```
Comment (danielschloms) [inline Sts1CobcSw/Periphery/Eps.cpp:10]: Remove periphery from namespace (see #139)

File: Sts1CobcSw/Periphery/Eps.hpp [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
```
Comment (PatrickKa) [inline Sts1CobcSw/Periphery/Eps.hpp:2]: 2 blank lines after the `#inlcude` block, even if it's empty.

File: Sts1CobcSw/Periphery/Eps.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
+#include <Sts1CobcSw/Hal/GpioPin.hpp>
+#include <Sts1CobcSw/Hal/IoNames.hpp>
+#include <Sts1CobcSw/Hal/Spi.hpp>
+#include <Sts1CobcSw/Periphery/Eps.hpp>
+#include <Sts1CobcSw/Serial/Byte.hpp>
+#include <Sts1CobcSw/Utility/Span.hpp>
+
+#include <rodos_no_using_namespace.h>
+
```
Comment (PatrickKa) [inline Sts1CobcSw/Periphery/Eps.cpp:9]: 2 blank lines after the `#inlcude` block

File: Sts1CobcSw/Periphery/Eps.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    // Changing to CNVST mode could be bad (analog input in digital pin)
+    constexpr auto clockMode = 0b10_b << 4;
+    constexpr auto referenceMode = 0b00_b;
+    auto setupData = 0_b | setupRegister | clockMode | referenceMode;
+    adcCsPin->Reset();
+    hal::WriteTo(&epsSpi, Span(setupData));
+    adcCsPin->Set();
+}
+}
```
Comment (PatrickKa) [inline Sts1CobcSw/Periphery/Eps.cpp:137]: Files must end with a blank line. Unfortunately, only clang-format 16 adds an option for that (https://releases.llvm.org/16.0.0/tools/clang/docs/ClangFormatStyleOptions.html#insertnewlineateof) but we still use clang-format 15.

File: Sts1CobcSw/Periphery/Eps.hpp [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+namespace sts1cobcsw::eps
+{
+auto Initialize() -> void;
+}
```
Comment (PatrickKa) [inline Sts1CobcSw/Periphery/Eps.hpp:6]: Files must end with a blank line.

**Review by PatrickKa — CHANGES_REQUESTED** [review]

File: Sts1CobcSw/Periphery/Eps.cpp [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    //      Don't use CNVST modes, we use the analog configuration for the CNVST pin
+    //      Therefore clock mode = 0b10 << 4
+    // [3:2]: Reference mode configuration
+    //      Reference off after scan; need wake-up delay: 0b00
+    //      Reference always on; no wake-up delay: 0b10
+    // [1:0]: Don't care
+
+    // Changing to CNVST mode could be bad (analog input in digital pin)
+    constexpr auto clockMode = 0b10_b << 4;
+    constexpr auto referenceMode = 0b00_b;
```
[the remark this answers is not in the document]
Comment (danielschloms) [reply Sts1CobcSw/Periphery/Eps.cpp:127]: If it's zero it wouldn't matter, but I have rewritten it in a clearer way.

**Review by PatrickKa — APPROVED** [review]

**Pull request merged**